torch/utils/_pytree.py

```diff
@@ -684,16 +684,16 @@ def _is_leaf(tree: PyTree, is_leaf: Optional[Callable[[PyTree], bool]] = None) -
 class TreeSpec:
     type: Any
     context: Context
-    children_specs: List["TreeSpec"]
+    _children_specs: List["TreeSpec"]
```

Comment:
```suggestion
    _children_specs: List[Self]
```
Or does dataclasses not like this typing_extension?